flink-ml-iteration/flink-ml-iteration-common/src/main/java/org/apache/flink/iteration/datacache/nonkeyed/ListStateWithCache.java

```diff
@@ -62,14 +63,32 @@ public class ListStateWithCache<T> implements ListState<T> {
     /** The data cache writer for the received records. */
     private final DataCacheWriter<T> dataCacheWriter;
 
-    @SuppressWarnings("unchecked")
     public ListStateWithCache(
             TypeSerializer<T> serializer,
             StreamTask<?, ?> containingTask,
             StreamingRuntimeContext runtimeContext,
             StateInitializationContext stateInitializationContext,
             OperatorID operatorID)
             throws IOException {
+        this(
+                serializer,
+                containingTask,
+                runtimeContext,
+                stateInitializationContext,
+                operatorID,
+                0.);
```

Comment:
With this constructor, the current algorithms using `ListStateWithCache` like `Kmeans` would not use managed memory at all, which causes performance regression.

Would it be better if we divide the memory evenly if multiple `ListStateWithCache` is used by default?

Reply:
Would it be better to set the default value as 1.0?

Reply:
@jiangxin369 and @zhipeng93 , thanks for your comments. 

After offline discussion, I found it is necessary to do check about sub-fractions from all usages on the operator-scope managed memory. So I changed the constructor of `ListStateWithCache`.

Please check the latest changes.